### PR TITLE
checkLinking: adding assertions

### PR DESCRIPTION
**Pull request by grewered**
- [ ] checkLinking

no checks

- [x] _checkOrCreateBookletColumn_
- [x] _simDat_
- [x] _checkLink_

### CODE REVIEW

File: R/checkLinking.r [hunk 1]
```diff
@@ -1,100 +1,161 @@
...
-          stopifnot(length(intersect(unlist(design), blocks)) > 0)
-          notInDe<- setdiff(blocks,unique(desL[,"blockName"]))
-          if(length(notInDe)>0) {warning(paste0(length(notInDe), " elements of 'blocks' vector missing in design."))}
-          desL   <- desL[which(desL[,"blockName"] %in% blocks),]                ### geht einfacher zu selektieren
+### ----------------------------------------------------------------------------
+### checks whether design is linked. checkLinking(design[1:15,c(1:5,ncol(design))], bookletColumn = "TH")
+
+checkLinking <- function(design, blocks=NULL, bookletColumn=NULL, verbose=FALSE){
+  # checks
+  checkmate::assert_data_frame(design)
```
Comment (weirichs) [inline R/checkLinking.r]: I think `checkmate::assert_data_frame(design)` isn't necessary as `design <- eatTools::makeDataFrame(design)` is called subsequently.

**Comment by grewered** [comment]
Do the other functions need checks? 
I'm not sure how the namespace thing works. How do I look at that?

File: R/checkLinking.r [hunk 2]
```diff
@@ -1,100 +1,160 @@
...
-          stopifnot(length(intersect(unlist(design), blocks)) > 0)
-          notInDe<- setdiff(blocks,unique(desL[,"blockName"]))
-          if(length(notInDe)>0) {warning(paste0(length(notInDe), " elements of 'blocks' vector missing in design."))}
-          desL   <- desL[which(desL[,"blockName"] %in% blocks),]                ### geht einfacher zu selektieren
+### ----------------------------------------------------------------------------
+### checks whether design is linked. checkLinking(design[1:15,c(1:5,ncol(design))], bookletColumn = "TH")
+
+checkLinking <- function(design, blocks=NULL, bookletColumn=NULL, verbose=FALSE){
+  # checks
+  checkmate::assert_character(blocks, null.ok = TRUE)
```
Comment (weirichs) [inline R/checkLinking.r]: I think, adding `unique = TRUE`, i.e. `checkmate::assert_character(blocks, null.ok = TRUE, unique = TRUE)` might be reasonable.

File: R/checkLinking.r [hunk 3]
```diff
@@ -1,100 +1,160 @@
...
+  checkmate::assert_vector(bookletColumn, null.ok = TRUE)
```
Comment (weirichs) [inline R/checkLinking.r]: I think this assertion is expendable as the argument `bookletColumns` is passed to the `eatTools::existsBackgroundVariables()` function, where the corresponding checks already take place. For reasons of economy, this line could be omitted.

**Review by weirichs — COMMENTED** [review]
Most looking very good, see few comments below.

**Comment by weirichs** [comment]
> Do the other functions need checks? I'm not sure how the namespace thing works. How do I look at that?

No, the other functions don't need assertion. If you are unsure, the NAMESPACE file tells you which functions are exported to the namespace, i.e. which functions can be called directly by the user. Only these functions usually require assertions. In the NAMESPACE, you can find `export(checkLinking)` but `export(checkLink)` is missing, for example. So `checkLink()` is invisible for the user and this function is only called internally by other functions. Assertions are not necessary here. 

And BTW: Only functions exported to the namespace need documentation.

**Review by weirichs — APPROVED** [review]
Thanks, LGTM

**Pull request merged**